src/hooks/useCreateEventAttestation.ts

```diff
@@ -0,0 +1,102 @@
+import { useSigner } from "@/hooks/useSigner";
+import { encodePacked, keccak256, zeroAddress } from "viem";
+import { EAS, SchemaEncoder } from "@ethereum-attestation-service/eas-sdk";
+import { useCallback, useEffect, useState } from "react";
+import type { Address } from "viem";
+import {
+  CREATE_EVENT_SCHEMA,
+  CREATE_EVENT_SCHEMA_UID,
+} from "@/components/CreateEventSchemaButton";
+
+export interface Event {
+  owner: Address;
+  name: string;
+  briefDescription: string;
+  fullDescription: string;
+  startsAt: number;
+  endsAt: number;
+  imageUrl: string;
+}
+
+export const CREATE_EVENT_SCHEMA_ENCODER = new SchemaEncoder(
+  CREATE_EVENT_SCHEMA
+);
+
+export const useCreateEventAttestation = () => {
+  // Address only valid for Optimism Sepolia
```

Comment:
🤔 think they all the same at least for OP stack networks since IIRC this contract comes deployed with the chain

Reply:
well observed